app/goal_routes.py

```diff
@@ -0,0 +1,97 @@
+from app import db
+from app.models.goal import Goal
+from app.models.task import Task
+from flask import Blueprint, jsonify, abort, make_response, request
+
+goal_bp = Blueprint("goals", __name__, url_prefix="/goals")
+
+def validate_model(cls, model_id):
+    try:
+        model_id = int(model_id)   
+    except:
+        abort(make_response({"details":"Invalid data"}, 400)) 
+    model = cls.query.get(model_id)
+    if not model:
+        abort(make_response({"message":f"{cls.__name__} {model_id} not found"}, 404))
+
+    return model 
+
+@goal_bp.route("", methods=["POST"])
+def create_goal():
+    try:
+        request_body = request.get_json()
+        new_goal = Goal.from_dict(request_body)
+    except:
+        abort(make_response({"details":"Invalid data"}, 400))
+
+    db.session.add(new_goal)
+    db.session.commit()
+
+    return make_response(jsonify({
+            "goal": Goal.to_dict(new_goal)})), 201
+
+@goal_bp.route("", methods=["GET"])
+def read_all_goals():
+    goals_query = Goal.query
+    goals = goals_query.all()
```

Comment:
Since you're not handling query params from the request to add additional query filters here (like sort by asc or desc), you acn directly call `Goal.query.all()` without needing to create the variable goals_query. 

However if you do choose to have filtering logic, then what you have is a good approach!